README.md

```diff
@@ -243,6 +243,150 @@ There are many differences between SQL syntax and functions supported by differe
 
 In order to use these functions, you need to explicitly import their corresponding modules.
 
+### Unsafe functions, operators and values
+
+Not all RDBMSs specific functions and operators are expose on each RDBMS module, there is
+also the problem of calling user defined functions. To overcome this problem Esqueleto
+exports a number of unsafe functions to call any function, operator or value. This functions
+can be found on Database.Esqueleto.Internal.Sql module.
+
+Warning: the functions discuss on section should always be use with an explicit type signature
+as they are too general in order to maintain type safety, and you will need to indicate some
+type when you want to get results back from the database so better set types earlier rather 
+than late. This functions are also vulnerable to sql injection attacks so their use must be
+limited and with care, more on this later.
+
+The most common use of this functions is for calling RDBMS specific or custom functions,
+for that end we use `unsafeSqlFunction`. For example, if we wish to consult the postgres
+`now` function we could so as follow:
+
+```haskell
+postgresTime :: (MonadIO m, MonadLogger m) => SqlWriteT m UTCTime
+postgresTime = 
+  select (return now) >>= return . head
+  where
+    now :: SqlExpr (Value UTCTime) 
+    now = unsafeSqlFunction (fromText "now") ([] :: [SqlExpr (Value Int)])
+```
+
+which generates this SQL:
+
+```sql
+SELECT now()
+```
+
+With the `now` function we could now use the current time of the postgres RDBMS on any query.
+Do notice that `now` does not use any arguments, so we use an empty list cast to a valid
+`UnsafeSqlFunctionArgument` instance to represent no arguments. 
+
+Note: make () an instance of `UnsafeSqlFunctionArgument` for when no arguments are needed, this
+casting is quite dumb.
+
+We can also use `unsafeSqlFunction` for more complex functions with customs values using 
+`unsafeSqlValue` which turns any string into a sql value of whatever type we want, disclaimer:
+if you use it badly you will cause a runtime error. For example, say we want to try postgres'
+`date_part` function and get the day of a timestamp, we could use:
+
+```haskell
+postgresTimestampDay :: (MonadIO m, MonadLogger m) => SqlWriteT m Int
+postgresTimestampDay = 
+  select (return $ dayPart date) >>= return . head
+  where
+    dayPart :: SqlExpr (Value UTCTime) -> SqlExpr (Value Int) 
+    dayPart s = unsafeSqlFunction (fromText "date_part") (unsafeSqlValue (fromText "\'day\'") :: SqlExpr (Value String) ,s)
+    date :: SqlExpr (Value UTCTime)
+    date = unsafeSqlValue . fromText $ "TIMESTAMP \'2001-02-16 20:38:40\'"
```

Comment:
I think this would be a better example for `unsafeSqlCastAs` - this can also be written as:

```haskell
postgresTimestampDay :: (MonadIO m, MonadLogger m) => UTCTime -> SqlPersistT m Int
postgresTimestampDay time =
  head <$> select (return (dayPart time))
  where
    addTimestampCast :: SqlExpr (Value UTCTime) -> SqlExpr (Value UTCTime)
    addTimestampCast time = unsafeSqlCastAs "TIMESTAMP" time

    dayPart :: SqlExpr (Value UTCTime) -> SqlExpr (Value Int)
    dayPart t = unsafeSqlFunction "date_part" (val "day", addTimestampCast t)
```

Note that we don't need `unsafeSqlValue` for `day` because `val :: Text -> SqlExpr (Value Text)` works just fine here. But if we try to pass in the UTCTime without a timestamp, it gets confused:

```
λ> now <- getCurrentTime
λ> dexDB $ select $ pure (unsafeSqlFunction "date_part" (val "Day", val now))
[Debug#SQL] SELECT date_part(?, ?)
; [PersistText "Day",PersistUTCTime 2019-10-29 15:12:06.381663489 UTC]
[Error#dexter] An error occurred: [SqlError]: SqlError {sqlState = "42725", sqlExecStatus = FatalError, sqlErrorMsg = "function date_part(unknown, unknown) is not unique", sqlErrorDetail = "", sqlErrorHint = "Could not choose a best candidate function. You might need to add explicit type casts."}
*** Exception: SqlError {sqlState = "42725", sqlExecStatus = FatalError, sqlErrorMsg = "function date_part(unknown, unknown) is not unique", sqlErrorDetail = "", sqlErrorHint = "Could not choose a best candidate function. You might need to add explicit type casts."}

λ> dexDB $ select $ pure (unsafeSqlFunction "date_part" (val "Day", unsafeSqlCastAs "TIMESTAMP" (val now) :: SqlExpr (Value UTCTime)))
[Debug#SQL] SELECT date_part(?, CAST(? AS TIMESTAMP))
; [PersistText "Day",PersistUTCTime 2019-10-29 15:12:06.381663489 UTC]
[Value {unValue = 29.0}]
```

Reply:
I think a better example may be MySQL global variables: https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html These are explicitly *not* functions syntactically, and it wouldn't work to use `val` either - you have to input the text directly.

Reply:
there is an example of `unsafeSqlCastAs` next to this one, I am closing this suggestion